android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java

```diff
@@ -100,7 +88,7 @@ public boolean onCreateOptionsMenu(Menu menu) {
 
     @Override
     public boolean onQueryTextChange(String query) {
-        if (query != null) {
+        if (!TextUtils.isEmpty(query)) {
```

Comment:
Can you please explain the use of this check? I mean we are already checking that in the line 37 of the ViewModel. Now if that is empty , then we will get an empty list from the ViewModel and we are already clearing the results List and handeling Visibility in ```searchQuery``` method in the Activity.
I know you haven't written this code but i feel this is redundant and should be removed to keep the code simpler.
@iamareebjamal What do you think ?

Reply:
Yes, that was what I was saying, it should be inside ViewModel, and if it there, it shouldn't be here anyway